src/dags/bbga.py

```diff
@@ -0,0 +1,356 @@
+            "frequency",
+        ),
+        # We don't actually need a `cdc_id` derived column for this table. However to ensure we
+        # can call the `PgComparatorCDCOperator` on all tables with the same set of parameters,
+        # they all need a `cdc_id`.
+        "derived": {"cdc_id": FuncCols(hash_columns, ("variabele",))},
+    },
+    "bbga_latest_and_greatest": {
+        "dialect": bbga_dialect,
+        "fieldnames": (
+            "jaar",
+            "gebiedcode_15",
+            "indicator_definitie_id",
+            "waarde",
+        ),
+        # As this table has over 6 million rows, the `cdc_id` derived column becomes a necessity
+        # for us to be able to call `PgComparatorCDCOperator` with the incredibly faster
+        # `--pg-copy` option. That option requires a key column with an integer value. That is
+        # what we define here.
+        #
+        # The `id` column here is by virtue of an `identifier` property in the schema.
+        # Schema-tools automatically creates an `id` column of type VARCHAR for us to fill with
+        # a value derived from the columns specified in the `identifier` property.
+        "derived": {
+            "id": FuncCols(concat_columns, ("indicator_definitie_id", "jaar", "gebiedcode_15")),
+            "cdc_id": FuncCols(hash_columns, ("indicator_definitie_id", "jaar", "gebiedcode_15")),
+        },
+    },
+    "bbga_std_latest_and_greatest": {
+        "dialect": bbga_dialect,
+        "fieldnames": (
+            "jaar",
+            "indicator_definitie_id",
+            "gemiddelde",
+            "standaardafwijking",
+            "bron",
+        ),
+        # We don't actually need a `cdc_id` derived column for this table. However to ensure we
+        # can call the `PgComparatorCDCOperator` on all tables with the same set of parameters,
+        # they all need a `cdc_id`.
+        #
+        # The `id` column here is by virtue of an `identifier` property in the schema.
+        # Schema-tools automatically creates an `id` column of type VARCHAR for us to fill with
+        # a value derived from the columns specified in the `identifier` property.
+        "derived": {
+            "id": FuncCols(concat_columns, ("indicator_definitie_id", "jaar")),
+            "cdc_id": FuncCols(hash_columns, ("indicator_definitie_id", "jaar")),
+        },
+    },
+}
+
+assert set(table_mappings.keys()) == set(
+    CSV_DEFINITIONS.keys()
+), "Both mappings should have the same set of keys."
+
+with DAG(dag_id=DAG_ID, default_args=default_args) as dag:
+    slack_at_start = MessageOperator(
+        task_id="slack_at_start",
+        http_conn_id="slack",
+        webhook_token=slack_webhook_token,
+        message=f"Starting {DAG_ID} ({DATAPUNT_ENVIRONMENT})",
+        username="admin",
+    )
+
+    mk_tmp_dir = PythonOperator(
+        # The resulting tmp dir is returned via XCom using the task_id
+        task_id="mk_tmp_dir",
+        python_callable=tempfile.mkdtemp,
```

Comment:
Het is erg handig voor debugging dat we DAG data in gedeelde folder liggen hebben, maar nu elke gebrokken run van deze DAG maakt eigen dir en die wordt nooit opgeschoond. 
Misschien toch gewoon net als alle andere DAGs `{TMP_DIR_BASE}/{dag_id}/` gebruiken?

Reply:
The idea behind a specific tmp directory per DAG run is to provide a form of isolation. You don't want leftovers from a previous run to potentially interfere with the next run. 

Imagine the following scenario:
- For some reason the CSV downloads stop working without failing the task and hence without failing the DAG
- Because we reuse the tmp directory from the previous run, we still find the previously downloaded CSV files.
- The DAG runs to completion, because it was able to process the (old) CSV files
- Without a failing DAG, we will not be notified of an error (that fact that we can't download CSV files anymore).

With a separate tmp directory per DAG run, this can never happen. You might argue that a failing download will _currently_ fail the task and hence the DAG. But who knows what bugs are introduced in the future. Maybe a carelessly inserted catch all `except` clause suddenly silences an error that would have previously failed the `HttpFetchOperator`. Isolation is good!

It also aids with debugging. The current setup ensure that what we find in a specific tmp directory is guaranteed to be from one specific run. 

If left around tmp directories are an issue, one could argue that it is the OS' responsibility to clean those up (at least those in `/tmp`)

Reply:
>> For some reason the CSV downloads stop working without failing the task and hence without failing the DAG
>> Because we reuse the tmp directory from the previous run, we still find the previously downloaded CSV files.

That's why we have last step to remove tmp dir if DAG is completed successfully.
Therefore separate tmp per run is not needed.

Reply:
> If left around tmp directories are an issue, one could argue that it is the OS' responsibility to clean those up (at least those in /tmp)

`/tmp` is used only in current situation, as we're running everything on single machine.
K8s uses `/airflow_tmp`, which is nog going to be cleaned (ever, I'm afraid).

And even if it was `/tmp` everywhere - OS/any other cleanup script has no understanding of which directory can or can not be removed, while DAG knows exactly when directory is not needed anymore.

Reply:
Had a chat with @ngaranko on the phone. Will use fixed tmp dir. Primarily to prevent build up of leftover cruft in object storage (future storage mechanism for tmp dirs versus current ephemeral pod tmp dirs)